users/users.go

```diff
@@ -24,6 +24,12 @@ var (
 type User struct {
 	Email    string
 	Password string
+	Metadata map[string]interface{}
+}
+
+type UserInfo struct {
```

Comment:
I just realized I think that you don't need this struct. Just use `User` without a password. On this layer, we always use entity object (in this case `User`).